From fd5d075b209b8e29fc0426ae2f22bf8d5358da78 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 10 May 2022 17:12:36 +0200
Subject: [PATCH 172/954] feat(stm32mp2): add UART driver for programmer

Add the compilation of UART driver, that will be used when programming
over UART.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d0450fc41d07d7d329d667b582473c2d528ce284 feat(stm32mp2): add UART driver for programmer]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Iad91be72084ae44c9a393cafb6924afe3cb6fe6e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/251632
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp2/platform.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 8eabd08207..2969732986 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -171,6 +171,10 @@ ifneq ($(filter 1,${STM32MP_RAW_NAND} ${STM32MP_SPI_NAND} ${STM32MP_SPI_NOR}),)
 BL2_SOURCES			+=	plat/st/stm32mp2/stm32mp2_boot_device.c
 endif
 
+ifeq (${STM32MP_UART_PROGRAMMER},1)
+BL2_SOURCES			+=	drivers/st/uart/stm32_uart.c
+endif
+
 BL2_SOURCES			+=	drivers/st/ddr/stm32mp2_ddr.c				\
 					drivers/st/ddr/stm32mp2_ddr_helpers.c			\
 					drivers/st/ddr/stm32mp2_ram.c
-- 
2.53.0

